threeML/classicMLE/joint_likelihood.py

```diff
@@ -790,11 +793,16 @@ def worker(start_index):
 
                 idx = cc.argmin()
 
-                print(
+                log.warning(
                     "Found a better minimum: %s with %s = %s. Run again your fit starting from here."
                     % (cc.min(), param_1, a[idx])
                 )
-
+        
+        else:
+            #restore model
+            for param in self._likelihood_model.parameters:
+                self._likelihood_model.parameters[param].value = model_clone[param].value
+        
```

Comment:
I'm curious, if the contour algorithm finds a better minimum, should we not run this? Also, how about instead using the built in "restore_best_fit" method?

Reply:
If we find a better minimum, I'm thinking we should keep the current state (new minimum) so that the user can choose to run `fit()` again from there (in interactive analyses).

So, `restore_best_fit()` is already called but I don't think it's what we want here: If there are more free parameters than dimensions in the likelihood profile, we actually perform a fit for every step in the profile. So `restore_best_fit` will restore the last minimum (best fit in the last profile step) rather than the results from the global fit.